### PR TITLE
finished Part1

### PR DESCRIPTION
**Pull request by xdie44**

### CODE REVIEW

**Comment by xdie44** [comment]
I couldn't clear up the warning :"padding structure...."
How can I do that?

File: main.cpp [hunk 1]
```diff
@@ -37,26 +37,360 @@ write 3 UDTs below (where it asks for Copied UDTs) that EACH have:
...
+        ~OscWave();
+
+        float duty, weight;
+        int intensity = 3;
+        bool play = false;
+        char type;
+
+        void StartPlaying(bool isPlaying);
+        void MakeNoise(float Strength, float weight);
+        int ChangeType(int whichOsc, int newType, int numOsc);
```
Comment (matkatmusic) [inline main.cpp:64]: member functions should start with a lowercase letter.  Static functions should start with a capital letter.
- [ ] revise

File: main.cpp [hunk 2]
```diff
@@ -37,26 +37,360 @@ write 3 UDTs below (where it asks for Copied UDTs) that EACH have:
...
+    float freq;
+    float volume {0.5};
+    bool oscCanPlay{true};
+    int numHarmonics;
+    struct OscWave 
+    {
+        OscWave();
+        ~OscWave();
+
+        float duty, weight;
```
Comment (matkatmusic) [inline main.cpp:57]: initialize these in-class, since your constructor doesn't do anything.
- [ ] revise

File: main.cpp [hunk 3]
```diff
@@ -37,26 +37,360 @@ write 3 UDTs below (where it asks for Copied UDTs) that EACH have:
...
+        int intensity = 3;
+        bool play = false;
+        char type;
```
Comment (matkatmusic) [inline main.cpp:60]: initialize in-class
- [ ] revise

File: main.cpp [hunk 4]
```diff
@@ -37,26 +37,360 @@ write 3 UDTs below (where it asks for Copied UDTs) that EACH have:
...
+    for (int i = 0 ; i < nHarm ; i++)
+    {
+        freqIn += freqIn + 5;
+    }
+    return freqIn;
+}
+
+float Oscillator::freqHalfTone (float freqTemp)
+{
+    return 1.1225f*freqTemp;
```
Comment (matkatmusic) [inline main.cpp:111]: improve the readability.  add spaces around your `*`
- [ ] revise

File: main.cpp [hunk 5]
```diff
@@ -37,26 +37,360 @@ write 3 UDTs below (where it asks for Copied UDTs) that EACH have:
...
+
+}
+
+char Oscillator::changeTimbre (char oscType1, char oscType2, bool isTheSame)
+{
+    if (isTheSame) return oscType1;
+    return oscType2;
```
Comment (matkatmusic) [inline main.cpp:118]: put `if` bodies on their own line.
also, you can write this with a ternary expression:
```
return isTheSame ? oscType1 : oscType2;
```

File: main.cpp [hunk 6]
```diff
@@ -37,26 +37,360 @@ write 3 UDTs below (where it asks for Copied UDTs) that EACH have:
...
+    float centerFreq;
+    float slope = 4.5f;
+    float resonance = 2.4f;
+
+    Filter();
+    ~Filter();
+
+    void FilterFreq (float freq, int nHarm);
+    void SwitchOffFilter (bool isFilterActive, int FilterT = 3);
+    int  nHarm (float freq, float freqMax = 20000.0f);
```
Comment (matkatmusic) [inline main.cpp:175]: capitalization of member function names.
- [ ] revise

File: main.cpp [hunk 7]
```diff
@@ -37,26 +37,360 @@ write 3 UDTs below (where it asks for Copied UDTs) that EACH have:
...
+void Filter::FilterFreq (float freq, int nHarm)
+{
+    int i = 0;
+    while (i < nHarm)
+    {
+            std::cout << "Freq n " << i << " " << freq * i;
+            i ++;
+    }
+
+}
```
Comment (matkatmusic) [inline main.cpp:198]: fix the indentation here.  the style guide in the readme.md explains what it should be set to.
- [ ] revise.

File: main.cpp [hunk 8]
```diff
@@ -37,26 +37,360 @@ write 3 UDTs below (where it asks for Copied UDTs) that EACH have:
...
+void Filter::SwitchOffFilter (bool isFilterActive, int filterT)
+{
+    int i = 1;
+    while (isFilterActive)
+        {
+            if (filterT == i) isFilterActive = false;
+             std::cout << "Filter n " << i << " is switched off!\n";
+            i++;
+        }
+    filterType = 'o';
```
Comment (matkatmusic) [inline main.cpp:209]: - [ ] revise, coding style.

File: main.cpp [hunk 9]
```diff
@@ -37,26 +37,360 @@ write 3 UDTs below (where it asks for Copied UDTs) that EACH have:
...
+int Filter::nHarm (float freq, float freqMax)
+{
+    int i = 1;
+    int j = 0;
+
+    while (freq * i < freqMax)
+    {
+        if (freq * i > 500) j++;
+        i++;
+    }
```
Comment (matkatmusic) [inline main.cpp:222]: put `if` bodies on their own line
- [ ] revise.

File: main.cpp [hunk 10]
```diff
@@ -37,26 +37,360 @@ write 3 UDTs below (where it asks for Copied UDTs) that EACH have:
...
+        ~LfoType();
+
+        float lfoDuty, lfoWeight;
+        int lfoIntensity = 3;
+        bool lfoPlay = false;
+        std::string type;
+
+        void StartPlaying(bool isPlaying);
+        void MakeNoise(float strength, float weight);
+        int ChangeType(int whichLfo, int newType, int numLfo);
```
Comment (matkatmusic) [inline main.cpp:251]: - [ ] revise.

File: main.cpp [hunk 11]
```diff
@@ -37,26 +37,360 @@ write 3 UDTs below (where it asks for Copied UDTs) that EACH have:
...
+    };
+
+    LfoType LfoT;
+
+    Lfo();
+    ~Lfo();
+
+    bool CanSwitchOnLfo (bool isLfoActive = true);
+    float ChangeIntensity (int lfoT, float intensityLfoIn, float intensityLfoOut);
+    char ChangeLfoWave (char waveTypeIn = 'a', char waveTypeOut = 'b');
```
Comment (matkatmusic) [inline main.cpp:262]: - [ ] revise.

File: main.cpp [hunk 12]
```diff
@@ -37,26 +37,360 @@ write 3 UDTs below (where it asks for Copied UDTs) that EACH have:
...
+    {
+        std::cout << "Now is playing\n";
+        isPlaying = true;
+    }
+}
+
+bool Lfo::CanSwitchOnLfo (bool isLfoActive)
+{
+    if (isLfoActive) return false;
+    return true;
```
Comment (matkatmusic) [inline main.cpp:326]: - [ ] revise.

File: main.cpp [hunk 13]
```diff
@@ -37,26 +37,360 @@ write 3 UDTs below (where it asks for Copied UDTs) that EACH have:
...
+        std::cout << "LFO n " << i << " " << intensityLfoOut << "\n";
+        intensityLfoOut += intensityLfoIn;
+    }
+    return intensityLfoOut;
+}
+
+char Lfo::ChangeLfoWave (char waveTypeIn, char waveTypeOut)
+{
+    if (waveTypeIn != 'o') return waveTypeOut;
+    return waveTypeIn;
```
Comment (matkatmusic) [inline main.cpp:342]: - [ ] revise.

File: main.cpp [hunk 14]
```diff
@@ -72,5 +406,56 @@ If you need inspiration for what to write, take a look at previously approved st
...
+
+    if (synth1.isOn) std::cout << "Synth 1 is On" << std::endl;
+    else std::cout << "Synth 1 is Off" << std::endl;
+    if (synth2.isOn) std::cout << "Synth 2 is On" << std::endl;
+    else std::cout << "Synth 2 is Off" << std::endl;
+    if (mainMod.isOn) std::cout << "Main modulator is On" << std::endl;
+    else std::cout << "Main modulator is Off" << std::endl;
+    if (secMod.isOn) std::cout << "Secondary modulator is On" << std::endl;
+    else std::cout << "Secondary modulator is Off" << std::endl;
+     std::cout << std::endl;
```
Comment (matkatmusic) [inline main.cpp:425]: group your if/else expressions by adding a blank line between them.  this will improve the readability.

- [ ] revise.

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp [hunk 15]
```diff
@@ -37,26 +37,360 @@ write 3 UDTs below (where it asks for Copied UDTs) that EACH have:
...
+}
+
+void Oscillator::OscWave::makeNoise (float strength, float weightT)
+{
+    if(weightT > strength)
+    {
+        std::cout << "OscWeight: " << weight << " > " << "strength: " << strength << std::endl;
+        std::cout << "Make Noise!\n";
+    }
+    else std::cout << "No Noise\n";
```
Comment (matkatmusic) [inline main.cpp:142]: If you're going to use curly braces for the `if` block, use them for the `else` block as well.
It's easy to miss the `else` as part of a larger logic structure **at first glance** when you leave out the curly braces.   The code ends up looking generically like this when you leave them out:
```
if( expr )
{
   statements;
}
more statements;
```
compare that with:
```
if( expr )
{
   statements;
}
else
{
    statements;
}
```

consistency is key to making your code as readable as possible.
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**